Feature/wormhole

deepak-likes-code opened the pull request:
# Pull Request Description

## Related Issue
Fixes # (issue number)

## Changes Made
- Added Wormhole token bridge integration
- Implemented cross-chain token transfer functionality
- Added support for creating wrapped tokens across chains 
  
## Implementation Details
- Implemented createWrappedTokenAction to allow creating wrapped versions of tokens from other chains
- Implemented tokenTransferAction to enable cross-chain token transfers via Wormhole
- Added support for retrieving supported chains with getWormholeSupportedChainsAction
- Integrated with Wormhole's token bridge protocol for secure cross-chain transfers
 

## Transaction executed by agent 
Example prompts and wormhole transactions: 

```Transfer 0.1 sol to BaseSepolia on testnet```

https://wormholescan.io/#/tx/0x1ccd55f3d2a2628410c3e8cdd39c8eb5239671f0d948ef4f425737f049d586fa?network=Testnet


```Transfer 0.01 Sol to Sui on main net```

https://wormholescan.io/#/tx/3AwGN8HWQC4X8d5T6jSXja3RBo8cqdz7cZ2RTKMyvELcUXztEqU1EzSmdXpJ5LTT63Pu6bqaSSZLPJr1mNfzw6wJ?network=Mainnet


```Transfer 1 USDC to base (mainnet)```

https://wormholescan.io/#/tx/5P6XzyVRHLkkZ2r9YUiJWe9VLWYvM3XqfM28YGfAR9Ey5Ys3eSEmDHNpNa9mTKNJA2u5zjVgwzVnsRfjSQAGg2cN?network=Mainnet

```Create a wrapped token of usdc on BaseSepolia```

https://wormholescan.io/#/tx/2Keh8Hsyx2bRxZxuyyBAnrxsbzYs31NXYa6Zod9GAW3YdBK2QFwszRY3ewKvZGVuH8ZayyTFnV7S8nW7Sou5pe8Z?network=Testnet&view=overview



## Additional Notes
This integration enables users to seamlessly move tokens between Solana and other blockchains supported by Wormhole, enhancing the cross-chain capabilities of the agent kit.

## Checklist
- [x] I have tested these changes locally
- [x] I have updated the documentation
- [x] I have added a transaction link
- [x] I have added the prompt used to test it

michaelessiet commented on src/actions/wormhole/cctpTransfer.ts:
```diff
@@ -0,0 +1,54 @@
+  schema: z.object({
+    destinationChain: z.string(),
+    transferAmount: z.string(),
+    network: z.string(),
```
Please make network optional and default to mainnet if network isn't provided

michaelessiet commented on src/actions/wormhole/createWrappedToken.ts:
```diff
@@ -0,0 +1,53 @@
+  schema: z.object({
+    destinationChain: z.string(),
+    tokenAddress: z.string(),
+    network: z.string(),
```
Please make this optional too. Reference prior comment

michaelessiet commented on src/actions/wormhole/tokenTransfer.ts:
```diff
@@ -0,0 +1,47 @@
+
+  schema: z.object({
+    destinationChain: z.string(),
+    network: z.string(),
```
Please make this optional too. Reference prior comment

deepak-likes-code replied:
@michaelessiet  fixed the above in the commit 03808f1d2b305deb1ff48f137122178b4fa679ae and the example prompts and its respective wormhole transaction links

michaelessiet commented:
@deepak-likes-code please can you add examples for the other actions you're adding in this PR as well

deepak-likes-code commented:
@michaelessiet I've added the examples for all the actions that i have implemented in the ReadMe , want to understand if you are talking about any other example

michaelessiet commented:
> @michaelessiet I've added the examples for all the actions that i have implemented in the ReadMe , want to understand if you are talking about any other example
> 
> 

No this looks good. Please run the lint command and fix any issues that it shows you, then this should be good to go

deepak-likes-code commented:
@michaelessiet fixed all the lint errors and we are good to go
